pytorch_lightning/trainer/training_loop.py

```diff
@@ -670,30 +637,28 @@ def run_training_batch(self, batch, batch_idx, dataloader_idx):
 
             if self.trainer.lightning_module.automatic_optimization:
                 for opt_idx, optimizer in self.get_active_optimizers(batch_idx):
-                    result = self._run_optimization(batch_idx, split_idx, split_batch, opt_idx, optimizer)
+                    result = self._run_optimization(batch_idx, split_batch, opt_idx, optimizer)
                     if result:
-                        batch_outputs[opt_idx].append(result.training_step_output_for_epoch_end)
+                        batch_outputs[opt_idx].append(result.training_step_output)
                         grad_norm_dict = result.get("grad_norm_dict", {})
             else:
                 # in manual optimization, there is no looping over optimizers
-                result = self._run_optimization(batch_idx, split_idx, split_batch)
+                result = self._run_optimization(batch_idx, split_batch)
                 if result:
-                    batch_outputs[0].append(result.training_step_output_for_epoch_end)
+                    batch_outputs[0].append(result.training_step_output)
 
-        output = AttributeDict(
+        return AttributeDict(
             signal=0,
-            # todo: Properly aggregate grad_norm accros opt_idx and split_idx
             grad_norm_dict=grad_norm_dict,
-            training_step_output_for_epoch_end=batch_outputs,
+            training_step_output=batch_outputs,
         )
-        return output
 
-    def _run_optimization(self, batch_idx, split_idx, split_batch, opt_idx=0, optimizer=None):
+    def _run_optimization(self, batch_idx, split_batch, opt_idx=0, optimizer=None):
         # TODO: In v1.5, when optimizer_idx gets removed from training_step in manual_optimization, change
         #   opt_idx=0 to opt_idx=None in the signature here
 
         # toggle model params + set info to logger_connector
-        self.run_train_split_start(split_idx, split_batch, opt_idx, optimizer)
+        self.run_train_split_start(batch_idx, split_batch, opt_idx, optimizer)
```

Comment:
this doesn't look right. accidental rename?

Reply:
@tchaton

Reply:
Oh yes, there is a bug. It should be `batch_idx * num_splits + split_idx`

Reply:
We just need to track the first ever sample seen per epoch to result the metrics.